game_patch/misc/vpackfile.cpp

```diff
@@ -291,27 +340,62 @@ static int vpackfile_build_file_list_new(const char* ext_filter, char*& filename
 
 static bool is_lookup_table_entry_override_allowed(rf::VPackfileEntry* old_entry, rf::VPackfileEntry* new_entry)
 {
+    // dirs are loaded in this order: base game, user_maps, client_mods, mods
     if (g_is_overriding_disabled) {
         // Don't allow overriding files after game is initialized because it can lead to crashes
         return false;
     }
-    if (!new_entry->parent->is_user_maps) {
-        // Allow overriding by packfiles from game root and from mods
+    // Allow overriding by packfiles from game root and from mods (not user_maps or client_mods)
+    else if (!new_entry->parent->is_user_maps && !new_entry->parent->is_client_mods) {
         return true;
     }
-    if (!old_entry->parent->is_user_maps && !stricmp(rf::file_get_ext(new_entry->name), ".tbl")) {
-        // Always skip overriding tbl files from game by user_maps
-        return false;
-    }
 #ifdef MOD_FILE_WHITELIST
-    if (is_mod_file_in_whitelist(new_entry->file_name)) {
+    else if (is_mod_file_in_whitelist(new_entry->name)) {
         // Always allow overriding for specific files
         return true;
     }
 #endif
-    if (!g_game_config.allow_overwrite_game_files) {
-        return false;
+    // Process if in client_mods
+    else if (new_entry->parent->is_client_mods) {
+        if (is_tbl_file_in_allowlist(new_entry->name) || is_tbl_file_a_hud_messages_file(new_entry->name)) {
+            // allow overriding of tbl files from client_mods if they are on allow list or are _text.tbl
+            g_is_modded_game = true; // goober todo: confirm what this is used for. If anticheat, can remove from here
```

Comment:
I don't think this is used anywhere currently. But yes, it was supposed to be something related to anticheat until I realized half of the playerbase has mods in user_maps